src/librustc_errors/json.rs

```diff
@@ -417,10 +417,16 @@ impl DiagnosticCode {
                 DiagnosticId::Error(s) => s,
                 DiagnosticId::Lint(s) => s,
             };
-            let explanation =
-                je.registry.as_ref().and_then(|registry| registry.find_description(&s));
+            let je_result = je
+                .registry
+                .as_ref()
+                .and_then(|registry| Some(registry.try_find_description(&s)))
+                .unwrap();
+            if je_result.is_ok() {
+                return DiagnosticCode { code: s, explanation: je_result.unwrap() };
+            }
 
-            DiagnosticCode { code: s, explanation }
+            DiagnosticCode { code: s, explanation: None }
```

Comment:
It looks like we can do `DiagnosticCode { code: s, explanation: je_result.ok() }` here actually.